File: 2058-concatenation-of-array/concatenation-of-array.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n;
        n = nums.size();
        //create ans of length 2n
        //ans[i] == nums[i]
        //ans[i+n] == nums[i]
        //0<=i<n
        //ans = concentation of 2 nums arrays
        //return ans
        for(int i = 0; i<2*n; i++) {
            if(i<n) {
                nums.insert(nums.begin() + i + n, nums[i]);
            }
        }
        vector<int> ans;
        ans = nums;
        return ans;
    }
};

File: 2058-concatenation-of-array/concatenation-of-array.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        # nums - length = n
        # ans - length = 2n
        # ans[i] = nums[i] and ans[i+n] == nums[i]
        # ans = 2 nums array concatenated
        # return ans
        n = len(nums)
        nums = nums + nums
        ans = nums
        return ans